Lib/fractions.py

```diff
@@ -234,10 +241,11 @@ def limit_denominator(self, max_denominator=1000000):
         if max_denominator < 1:
             raise ValueError("max_denominator should be at least 1")
         if self._denominator <= max_denominator:
-            return Fraction(self)
+            return Fraction(self._numerator, self._denominator, _normalize=False)
```

Comment:
Does this change make a significant difference to performance? If not, I'd suggest reverting it, both for readability and for the purposes of keeping the PR focused. From looking at the code, a simple `Fraction(self)` shouldn't end up doing any gcd calculation here.

Reply:
It actually saves 30% on this case based on tests above -- however, I hadn't realized at the time that the `return self` was executed in this branch and no normalization was done.  The speed increase is probably based on the `isinstance()` check, which just keeps getting faster and faster in subsequent versions, so it might eventually be the faster route.  I've removed the change.

If Fractions are eventually to be truly immutable and hashable, then `return self` is the correct response here, but that is probably backwards incompatible with real code where people change `_numerator` or `_denominator` after instantiation.